### DISCUSSION TITLE
Run as specific non-root user

### DISCUSSION THREAD

**Discussion by robflate** [sysadminsmedia/homebox#218] [Q&A]
How can I run Homebox as a specific non-root user rather than either `root` or `65532`? I want to run it as my home user `1000`. Some containers allow this with the env vars `PUID` and `PGID` and others use Dockers built in `user: PUID:PGID`. What is the suggested method? Thanks.

**Comment by tankerkiller125** [comment]
The rootless container build is using alpine as the base, with "nonroot" as the username. I think those environment variables might work? But honestly I don't have much experience with the docker build side of things for that.

**Comment by 101br03k** [comment]
> My finding is, that this is something unique to how linuxserver builds their images. It is described [here](https://docs.linuxserver.io/general/understanding-puid-and-pgid).

source
https://www.reddit.com/r/docker/comments/tshn82/how_does_puidpgid_actually_work_in_docker/

**Reply by tankerkiller125** [reply]
It looks like the `user` flag though is a docker global feature though.

Looks like in compose it would be something like `user: person` or `user: PID:GID` which should work across any container based on what I'm finding.

**Reply by 101br03k** [reply]
Only thing I found is this
https://docs.docker.com/engine/containers/run/#user

I made a copy of the compose file of my uptime kuma container and added the user flag. 
It has no rights on the files, the files are being made by root, even when removing them and letting the container recreate them by starting the container, it could be that my user is in the docker group and thas creates the files first as root?


version: "3.3"
services:
  uptime-kuma2:
    image: louislam/uptime-kuma:latest
    container_name: uptime-kuma2
    volumes:
      - ./uptime-kuma-data:/app/data
      - /var/run/docker.sock:/var/run/docker.sock
    ports:
      - 3002:3001 # <Host Port>:<Container Port>
    restart: always
    network_mode: bridge
    user: "1000"

![image](https://github.com/user-attachments/assets/c67c167e-9acf-4a8f-b2c0-a2de1cf1a65d)

**Reply by tankerkiller125** [reply]
Uptime Kuma likely doesn't have a rootless image (especially since it needs access to the docker.sock file).

I'll have to investigate Homebox with the user flag at a later point.
